File: python/scripts/list_invoice_profiles.py
import os
from travelperk_http_python.builder.builder import build
from utils import write_output
travelperk = build(os.environ["SANDBOX_API_KEY"], True)
profiles = travelperk.expenses().invoice_profiles().query().set_offset(1).set_limit(10).get()
write_output("profiles", [
    profiles.offset,
    profiles.limit,
    profiles.total,
    profiles.profiles,
    profiles.profiles[0].id,
    profiles.profiles[0].name,
    profiles.profiles[0].payment_method_type,
    profiles.profiles[0].billing_period,
    profiles.profiles[0].currency,
    profiles.profiles[0].billing_information.legal_name,
    profiles.profiles[0].billing_information.vat_number,
    profiles.profiles[0].billing_information.address_line_1,
    profiles.profiles[0].billing_information.address_line_2,
    profiles.profiles[0].billing_information.city,
    profiles.profiles[0].billing_information.postal_code,
    profiles.profiles[0].billing_information.country_name,
])

File: php/scripts/list_invoice_profiles.php
<?php
require_once(__DIR__."/../vendor/autoload.php");
require_once("utils.php");
use Namelivia\TravelPerk\ServiceProvider;
$travelperk = (new ServiceProvider())->build(getenv("SANDBOX_API_KEY"), true);
$profiles = $travelperk->expenses()->invoiceProfiles()->query()->setOffset(1)->setLimit(10)->get();
write_output("profiles", [
    $profiles->offset,
    $profiles->limit,
    $profiles->total,
    $profiles->profiles,
    $profiles->profiles[0]->id,
    $profiles->profiles[0]->name,
    $profiles->profiles[0]->paymentMethodType,
    $profiles->profiles[0]->billingPeriod,
    $profiles->profiles[0]->currency,
    $profiles->profiles[0]->billingInformation->legalName,
    $profiles->profiles[0]->billingInformation->vatNumber,
    $profiles->profiles[0]->billingInformation->addressLine1,
    $profiles->profiles[0]->billingInformation->addressLine2,
    $profiles->profiles[0]->billingInformation->city,
    $profiles->profiles[0]->billingInformation->postalCode,
    $profiles->profiles[0]->billingInformation->countryName,
]);
